packages/base-map/src/funcs.js

```diff
@@ -0,0 +1,6 @@
+export default function callIfValid(fn) {
```

Comment:
I'd prefer calling this file `utils.js`.

Reply:
Also, please add some jsDoc for this function.

Reply:
Do you think this could live in core-utils?

Reply:
Yeah, I think it makes sense to live in core-utils ... pretty generic (check if param is a function, and return that as a closure or return a closure that does nothing -- that's what it's doing, right?).

Reply:
Actually I was just thinking a little more about this. I think maybe a better way to go about this is to make an abstract overlay class with methods that can be overridden that each overlay can implement.

Reply:
Yep that is the next level, that would remove the need for the derived class to register itself.

Reply:
BTW, I was trying to use this with the vehicles code that I've been working on.  I refactored that code to use functional components, and couldn't get it to work.  So switched back to a class, and that kinda worked.

But there might be a pretty big performance issue here.  With ~500 markers (ala vehicles), turning a layer on & off is really slow.  From what I can tell, React-Leaflet makes every marker a separate layer.  Researching...